utils/db_utils.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime, timezone
+
+import pandas as pd
+
+import emission.core.get_database as edb

```

Comment:
Please provide the link to the Timeseries_sample notebook that you've mentioned.

Reply:
There are examples all over the e-mission codebase as well. Timeseries_sample (from the e-mission-server repo) has a high level overview, but you can just look at the classes in `emission/storage/timeseries` and `emission/storage/decorations`

Reply:
I couldn't find equivalent functions. If there are these functions, please give me the full path.

Reply:
@AlirezaRa94 couldn't find equivalent functions for which calls?
If you list out the calls, I can give you a mapping.

This is particularly important since the current methods in here are causing errors while being used with DocumentDB, a MongoDB-compatible-ish hosted solution that is used at NREL.

We test the timeseries functionality against DocumentDB, but arbitrary queries may or may not work.
We are not able to deploy the dashboard on staging because of this issue.

```
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/dash/_callback.py", line 447, in add_context
output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
File "app_sidebar_collapsible.py", line 190, in update_store_trips
df = query_confirmed_trips(start_date_obj, end_date_obj)
File "/usr/src/app/utils/db_utils.py", line 66, in query_confirmed_trips
df = pd.json_normalize(list(query_result))
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1207, in next
if len(self.__data) or self._refresh():
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1124, in _refresh
self.__send_message(q)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/cursor.py", line 1001, in __send_message
address=self.__address)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1372, in _run_operation_with_response
exhaust=exhaust)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1471, in _retryable_read
return func(session, server, sock_info, slave_ok)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/mongo_client.py", line 1366, in _cmd
unpack_res)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/server.py", line 137, in run_operation_with_response
first, sock_info.max_wire_version)
File "/root/miniconda-4.12.0/envs/emission/lib/python3.7/site-packages/pymongo/helpers.py", line 168, in _check_command_response
max_wire_version)
pymongo.errors.OperationFailure: Internal server error, full error: {'ok': 0.0, 'operationTime': Timestamp(1681425547, 1), 'code': 42, 'errmsg': 'Internal server error'}
```

Reply:
@AlirezaRa94 I took a quick look at the functions - they are for loading uuids and trip tables.

You can use TimeSeries.find_entries() to get a list of entries for a particular time range (TimeQuery), which you can then call `json_normalize` on and then select the columns (= apply the projection) using pandas.